### ISSUE TITLE
IntelliKeys USB Arduino Library available

### ISSUE THREAD

**Issue by gdsports** [ATMakersOrg/OpenIKeys#2]
I purchased an IntelliKeys (IK) USB device from Ebay hoping to use it as a large touch surface. The IK USB interface is vendor specific so it is lot more complex than I expected. Fortunately, I found this project.

The Teensy 3.6 https://www.pjrc.com/store/teensy36.html (t36) is an Arduino compatible board which works with the popular Arduino IDE. The t36 was selected because it has two USB ports so it can operate between an IK and a computer. The IK plugs into the t36 USB host port. The t36 plugs in to the computer using its other USB port. A far as the computer knows, it is connected to a USB keyboard and mouse so it does not need any special drivers or software.

So far the Teensy 3.6 gets touch input from an IK and sends USB keyboard and mouse output. The example included with the library is a limited QWERTY keyboard demo. This is not a full implementation of the OpenIKeys software. The library could be the foundation for such an effort. At this stage, the library is mostly of interest to software developers rather than IK users.

The IntelliKeys Teensy 3.6 library is on github at https://github.com/gdsports/IntelliKeys_t36

**Comment by ATMakersBill** [comment]
GD I’m so sorry I missed this thank you for the wonderful work!

**Comment by willwade** [comment]
This looks neat. Just a FYI too we've been looking at an Adafruit Feather in much the same way. We have an extra bonus that we can make it all Bluetooth..And we can use Proper C codes for keys and it allows an unlimited number of shifted keys.  *Very work in progress and a bit theoretical.* - nothing to show just yet but a similar project: https://github.com/AceCentre/RelayKeys

**Comment by gdsports** [comment]
@willwade

The latest Arduino IntelliKeys driver is now working on SAMD21/Trinket M0. https://github.com/gdsports/IntelliKeys_uhls

RelayKeys looks very interesting esp. the BLE keyboard and mouse support. This could make the IK usable with tablets and phones. Adafruit is adding BLE CircuitPython support to their nRF52840 boards
so this is yet another option.

**Comment by willwade** [comment]
Yeah. They’ve done a great job. Check out the new (yet to buy) official Arduino ble boards. From memory they have the same nrf chipset which is working really well.
